File: IPWorks ZIP Samples/Create SevenZip/java/createsevenzip.java
/*
 * IPWorks ZIP 2024 Java Edition - Sample Project
 *
 * This sample project demonstrates the usage of IPWorks ZIP in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/ipworkszip
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */

import java.io.*;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import ipworkszip.*;

public class createsevenzip extends ConsoleDemo {
  SevenZip sevenzip;

  public createsevenzip() {
    sevenzip = new SevenZip();

    try {
      sevenzip.setArchiveFile(prompt("Please enter the name of the 7z file to create", ":", "test.7z"));
      char ch = ask("Recurse Subdirectories");
      if (ch == 'n' || ch == 'N')
        sevenzip.setRecurseSubdirectories(false);

      sevenzip.includeFiles(prompt("Please enter the path of the directory to compress", ":", "./*"));
      
      System.out.println("Compressing...");
      sevenzip.compress();
      System.out.println("Directory compressed.");
    } catch (IPWorksZipException ex) {
      System.out.println("IPWorksZip exception thrown: " + ex.getCode() + " [" + ex.getMessage() + "].");
    } catch (Exception ex) {
      System.out.println(ex.getMessage());
    }
  }

  public static void main(String[] args) {
    new createsevenzip();
  }
}


class ConsoleDemo {
  private static BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

  static String input() {
    try {
      return bf.readLine();
    } catch (IOException ioe) {
      return "";
    }
  }
  static char read() {
    return input().charAt(0);
  }

  static String prompt(String label) {
    return prompt(label, ":");
  }
  static String prompt(String label, String punctuation) {
    System.out.print(label + punctuation + " ");
    return input();
  }
  static String prompt(String label, String punctuation, String defaultVal) {
      System.out.print(label + " [" + defaultVal + "]" + punctuation + " ");
      String response = input();
      if (response.equals(""))
        return defaultVal;
      else
        return response;
  }

  static char ask(String label) {
    return ask(label, "?");
  }
  static char ask(String label, String punctuation) {
    return ask(label, punctuation, "(y/n)");
  }
  static char ask(String label, String punctuation, String answers) {
    System.out.print(label + punctuation + " " + answers + " ");
    return Character.toLowerCase(read());
  }

  static void displayError(Exception e) {
    System.out.print("Error");
    if (e instanceof IPWorksZipException) {
      System.out.print(" (" + ((IPWorksZipException) e).getCode() + ")");
    }
    System.out.println(": " + e.getMessage());
    e.printStackTrace();
  }

  /**
   * Takes a list of switch arguments or name-value arguments and turns it into a map.
   */
  static java.util.Map<String, String> parseArgs(String[] args) {
    java.util.Map<String, String> map = new java.util.HashMap<String, String>();
    
    for (int i = 0; i < args.length; i++) {
      // Add a key to the map for each argument.
      if (args[i].startsWith("-")) {
        // If the next argument does NOT start with a "-" then it is a value.
        if (i + 1 < args.length && !args[i + 1].startsWith("-")) {
          // Save the value and skip the next entry in the list of arguments.
          map.put(args[i].toLowerCase().replaceFirst("^-+", ""), args[i + 1]);
          i++;
        } else {
          // If the next argument starts with a "-", then we assume the current one is a switch.
          map.put(args[i].toLowerCase().replaceFirst("^-+", ""), "");
        }
      } else {
        // If the argument does not start with a "-", store the argument based on the index.
        map.put(Integer.toString(i), args[i].toLowerCase());
      }
    }
    return map;
  }
}

File: IPWorks ZIP Samples/Create SevenZip/php/create_sevenzip.php
<?php
/*
 * IPWorks ZIP 2024 PHP Edition - Sample Project
 *
 * This sample project demonstrates the usage of IPWorks ZIP in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/ipworkszip
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */
require_once('../include/ipworkszip_sevenzip.php');
require_once('../include/ipworkszip_const.php');
?>
<?php
if ($argc < 3) {
  echo "Usage: php create_sevenzip.php name path [recurse]\n\n";
  echo "  name        the name of the 7z file to create\n";
  echo "  path        the path of the directory to compress\n";
  echo "  recurse     whether to recurse subdirectories (optional)\n";
  echo "\nExample: php create_sevenzip.php test.7z c:\\mydir recurse\n\n";
} else {
  try {
    $sevenzip = new IPWorksZip_SevenZip();

    $sevenzip->setArchiveFile($argv[1]);
    $sevenzip->setRecurseSubdirectories(in_array("recurse", $argv));
    $sevenzip->doIncludeFiles($argv[2]);

    echo "Compressing...\n";
    $sevenzip->doCompress();
    echo "Directory compressed.\n";
  } catch (Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
  }
}
?>